File: feature/base/src/main/java/net/twisterrob/android/view/color/swatches/pixel/drawer/LineByLineBitmapDrawer.java
package net.twisterrob.android.view.color.swatches.pixel.drawer;

import net.twisterrob.android.view.color.swatches.pixel.color.PixelColor;

public class LineByLineBitmapDrawer extends BitmapDrawer {
	public LineByLineBitmapDrawer(int[] bitmap, int w, int h, PixelColor pixel) {
		super(bitmap, w, h, pixel);
	}

	@Override
	protected void fillPixels() {
		int w = this.w;
		int h = this.h;
		for (int y = 0; y < h; ++y) {
			for (int x = 0; x < w; ++x) {
				bitmap[y * w + x] = pixel.getPixelColorAt(x, y);
			}
			reportProgress();
		}
	}

	public static Factory factory() {
		return new Factory() {
			@Override
			public BitmapDrawer create(int[] bitmap, int w, int h, PixelColor color) {
				return new LineByLineBitmapDrawer(bitmap, w, h, color);
			}
		};
	}
}

File: feature/base/src/main/java/net/twisterrob/android/view/color/swatches/pixel/drawer/LineByLineBitmapDrawer.kt
package net.twisterrob.android.view.color.swatches.pixel.drawer

import net.twisterrob.android.view.color.swatches.pixel.color.PixelColor

class LineByLineBitmapDrawer(
	bitmap: IntArray, w: Int, h: Int, pixel: PixelColor
) : BitmapDrawer(bitmap, w, h, pixel) {

	override fun fillPixels() {
		val w = this.w
		val h = this.h
		for (y in 0 until h) {
			for (x in 0 until w) {
				bitmap[y * w + x] = pixel.getPixelColorAt(x, y)
			}
			reportProgress()
		}
	}

	companion object {

		@JvmStatic
		fun factory() = object : BitmapDrawer.Factory {
			override fun create(bitmap: IntArray, w: Int, h: Int, pixel: PixelColor) =
				LineByLineBitmapDrawer(bitmap, w, h, pixel)
		}
	}
}
